src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingTimeSyncState.java

```diff
@@ -225,15 +232,19 @@ private long computeClusterTimeOffset(List<Long> timeSyncOffsets) {
         return Math.round(sum / (double) count);
     }
 
-    private void handleErroneousTimeSyncRequest(BeaconSendingContext context) {
+    private void handleErroneousTimeSyncRequest(TimeSyncResponse response, BeaconSendingContext context) {
 
         // if this is the initial sync try, we have to initialize the time provider
         // in every other case we keep the previous setting
         if (initialTimeSync) {
             context.initializeTimeSync(0, context.isTimeSyncSupported());
         }
 
-        if (context.isTimeSyncSupported()) {
+        if (response != null && response.getResponseCode() == Response.HTTP_TOO_MANY_REQUESTS) {
```

Comment:
is this the same as BeaconSendingRequestUtil.isTooManyRequestsResponse(statusResponse) in which case I would prefer the latter